src/instantiator.js

```diff
@@ -22,6 +22,23 @@ function isPrimitive(obj) {
 }
 
 /**
+ * Checks whether a property is on required array.
+ * @param property - the property to check.
+ * @param requiredArray - the required array
+ * @returns {boolean}
+ */
+function isPropertyRequired(property, requiredArray) {
+  var found = false;
+  var i = 0;
+  requiredArray = requiredArray || [];
+  while (i < requiredArray.length && !found) {
```

Comment:
Can you change the loop to a `forEach` one?
Also use an `if` instead of the next line, should be something like:

```javascript```
requiredArray.forEach(function(requiredProperty) {
  if (requiredProperty === property) {
    found = true;
  }
});
```

I find it to be more readable.